x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/components/create_analytics_form/create_analytics_form.tsx

```diff
@@ -4,47 +4,67 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { Fragment, FC } from 'react';
+import React, { Fragment, FC, useEffect } from 'react';
 
 import {
-  EuiCallOut,
   EuiComboBox,
   EuiForm,
   EuiFieldText,
   EuiFormRow,
   EuiLink,
-  EuiSpacer,
+  EuiRange,
   EuiSwitch,
-  EuiText,
 } from '@elastic/eui';
 
 import { i18n } from '@kbn/i18n';
-import { FormattedMessage } from '@kbn/i18n/react';
 
 import { metadata } from 'ui/metadata';
-import { INDEX_PATTERN_ILLEGAL_CHARACTERS } from 'ui/index_patterns';
+import { IndexPattern, INDEX_PATTERN_ILLEGAL_CHARACTERS } from 'ui/index_patterns';
+import { Field, EVENT_RATE_FIELD_ID } from '../../../../../../common/types/fields';
 
+import { newJobCapsService } from '../../../../../services/new_job_capabilities_service';
+import { useKibanaContext } from '../../../../../contexts/kibana';
 import { CreateAnalyticsFormProps } from '../../hooks/use_create_analytics_form';
+import { JOB_TYPES } from '../../hooks/use_create_analytics_form/state';
 import { JOB_ID_MAX_LENGTH } from '../../../../../../common/constants/validation';
+import { Messages } from './messages';
+import { JobType } from './job_type';
 
 // based on code used by `ui/index_patterns` internally
 // remove the space character from the list of illegal characters
 INDEX_PATTERN_ILLEGAL_CHARACTERS.pop();
 const characterList = INDEX_PATTERN_ILLEGAL_CHARACTERS.join(', ');
 
+const NUMERICAL_FIELD_TYPES = new Set([
```

Comment:
Maybe consider adding the values from the `ES_FIELD_TYPES` enum from the data plugin here (in src/plugins/data/common)?